lib/slax_web/websockets/disable.ex

```diff
@@ -0,0 +1,179 @@
+defmodule SlaxWeb.Disable do
+  @moduledoc """
+  A module that handles Slack websocket payloads for disabling Slax interactions and
+  builds modal views for slack https://api.slack.com/reference/surfaces/views
+  """
+
+  alias Slax.Slack
+  alias Slax.Channels
+
+  def handle_payload(%{
+        "trigger_id" => trigger_id,
+        "type" => "shortcut",
+        "callback_id" => "disable_slax"
+      }) do
+    view = build_disable_view(%{trigger_id: trigger_id})
+
+    Slack.open_modal(%{trigger_id: trigger_id, view: view})
+  end
+
+  def handle_payload(%{
+        "trigger_id" => trigger_id,
+        "type" => "shortcut",
+        "callback_id" => "enable_slax"
+      }) do
+    view = build_enable_view(%{trigger_id: trigger_id})
+
+    Slack.open_modal(%{trigger_id: trigger_id, view: view})
+  end
+
+  def handle_payload(%{
+        "trigger_id" => _trigger_id,
+        "type" => "view_submission",
+        "view" => %{
+          "callback_id" => "disable_view",
+          "state" => %{
+            "values" => values
+          }
+        }
+      }) do
+    with %{
+           "channel_select" => %{
+             "selected_option" => %{"text" => %{"text" => name}, "value" => channel_id}
+           }
+         } <-
+           parse_state_values(values) do
+      Channels.create_or_update_channel(channel_id, %{
+        name: name,
+        disabled: true
+      })
+
+      :ok
+    end
+  end
+
+  def handle_payload(%{
+        "trigger_id" => _trigger_id,
+        "type" => "view_submission",
+        "view" => %{
+          "callback_id" => "enable_view",
+          "state" => %{
+            "values" => values
+          }
+        }
+      }) do
+    with %{
+           "channel_select" => %{
+             "selected_option" => %{"text" => %{"text" => name}, "value" => channel_id}
+           }
+         } <-
+           parse_state_values(values) do
+      Channels.create_or_update_channel(channel_id, %{
+        name: name,
+        disabled: false
+      })
+
+      :ok
+    end
+  end
+
+  defp parse_state_values(values) do
+    values
+    |> Map.values()
+    |> Enum.reduce(%{}, fn value, acc -> Map.merge(acc, value) end)
+  end
```

Comment:
I believe it's doing what you need it to do, but handling the payload this way makes it really hard to understand the data structures we're working with. both the input and return value are mysterious and it's unclear what assumptions are being made about either side.

If possible, I think it might help to (a) use more explicit pattern matching on the input values and (b) return data that's structured in a specific way that's not just inherited from the input (e.g. it could return a map with keys for `channel_id` and `channel_name`)